docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,999 @@
+     ]
+    }
+   ],
+   "source": [
+    "# Relative entropy list\n",
+    "relative_entropy_values = []\n",
+    "# Generator loss list\n",
+    "generator_loss_values = []\n",
+    "# Discriminator loss list\n",
+    "discriminator_loss_values = []\n",
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    relative_entropy_epoch = []\n",
+    "    generator_loss_epoch = []\n",
+    "    discriminator_loss_epoch = []\n",
+    "    for i, data in enumerate(dataloader):\n",
+    "        # Adversarial ground truths\n",
+    "        valid = torch.ones(data.size(0), 1)\n",
+    "        fake = torch.zeros(data.size(0), 1)\n",
+    "\n",
+    "        # Generate a batch of data points\n",
+    "        gen_data = generator()\n",
+    "\n",
+    "        # Evaluate Relative Entropy\n",
+    "        relative_entropy_epoch.append(get_relative_entropy(gen_data))\n",
+    "\n",
+    "        # ---------------------\n",
+    "        #  Train Discriminator\n",
+    "        # ---------------------\n",
+    "        optimizer_disc.zero_grad()\n",
+    "\n",
+    "        # Loss measures discriminator's ability to distinguish real from generated samples\n",
+    "        disc_data = discriminator(data)\n",
+    "\n",
+    "        real_loss = disc_loss_fun(disc_data, valid)\n",
+    "        fake_loss = disc_loss_fun(discriminator(gen_data), fake)\n",
+    "        discriminator_loss = (real_loss + fake_loss) / 2\n",
+    "\n",
+    "        discriminator_loss.backward(retain_graph=True)\n",
+    "        optimizer_disc.step()\n",
+    "\n",
+    "        # -----------------\n",
+    "        #  Train Generator\n",
+    "        # -----------------\n",
+    "\n",
+    "        optimizer_gen.zero_grad()\n",
+    "\n",
+    "        # Loss measures generator's ability to prepare good data samples\n",
+    "        generator_loss = gen_loss_fun(discriminator(gen_data), valid)\n",
+    "        generator_loss.retain_grad = True\n",
+    "        g_loss_grad = generator_loss_grad(\n",
+    "            qc, twolocal.ordered_parameters, generator.weight.data.numpy(), discriminator\n",
+    "        )\n",
+    "\n",
+    "        # generator_loss.backward(retain_graph=True)\n",
+    "        for j, param in enumerate(generator.parameters()):\n",
+    "            param.grad = g_loss_grad\n",
+    "        optimizer_gen.step()\n",
+    "\n",
+    "        generator_loss_epoch.append(generator_loss.item())\n",
+    "        discriminator_loss_epoch.append(discriminator_loss.item())\n",
+    "\n",
+    "        print(\n",
+    "            \"[Epoch %d/%d] [Batch %d/%d] [D loss: %f] [G loss: %f]\"\n",
+    "            % (\n",
+    "                epoch,\n",
+    "                num_epochs,\n",
+    "                i,\n",
+    "                len(dataloader),\n",
+    "                discriminator_loss.item(),\n",
+    "                generator_loss.item(),\n",
+    "            )\n",
+    "        )\n",
+    "\n",
```

Comment:
It could be cool to add a callback that plots the losses as you train, instead of printing them. You could check out the code in this tutorial: https://github.com/Qiskit/qiskit-machine-learning/blob/main/docs/tutorials/02_neural_network_classifier_and_regressor.ipynb

(not necessary in any way, feel free to ignore the suggestion)

Reply:
Added a callback (although it is not a callback), now losses and entropy are plotted during the training process.